src/rpc/RPCHelpers.cpp

```diff
@@ -596,6 +592,39 @@ ledgerInfoFromRequest(Context const& ctx)
     return *lgrInfo;
 }
 
+// extract ledgerInfoFromRequest's parameter from context
+std::variant<Status, ripple::LedgerInfo>
+getLedgerInfoFromHashOrSeq(
+    BackendInterface const& backend,
+    boost::asio::yield_context& yield,
+    std::optional<std::string> ledgerHash,
+    std::optional<uint32_t> ledgerIndex,
+    uint32_t maxSeq)
+{
+    std::optional<ripple::LedgerInfo> lgrInfo;
+    auto const err =
+        RPC::Status{RPC::RippledError::rpcLGR_NOT_FOUND, "ledgerNotFound"};
+    if (ledgerHash)
+    {
+        lgrInfo =
+            backend.fetchLedgerByHash(ripple::uint256{*ledgerHash}, yield);
+        if (!lgrInfo || lgrInfo->seq > maxSeq)
+            return err;
+
+        return *lgrInfo;
+    }
+    const auto ledgerSequence = ledgerIndex.value_or(maxSeq);
```

Comment:
Let's not mix east and west const in one function (or better in the entire codebase) 😃